raiden/utils/upgrades.py

```diff
@@ -0,0 +1,60 @@
+import importlib
+import os
+import shutil
+from pathlib import Path
+
+from raiden.exceptions import RaidenDBUpgradeBackupError, RaidenDBUpgradeExecutionError
+from raiden.storage.serialize import JSONSerializer
+from raiden.storage.sqlite import SQLiteStorage
+
+
+class UpgradeManager:
```

Comment:
The problem that it is solving is migrating from version X to version Y. The list can't be hardcoded because you would want to skip say, V1 and V2 migrations if you're already running version 3. So you'd start from any migration from that point on.

Reply:
It's possible to skip with these approaches:

##### The migration itself does the check

```python
def upgrade_v1(data):
  if data['version'] != 'v0': return
  # do the migration only from v0 to v1
```

pros: simple & straightforward
cons: don't know

##### The list would have an entry for each version

```python
migrations = [
   upgrade_v1,
   empty,
   upgrade_v3,
]
migrations_to_apply = migrations[current_version:]
```

pros: don't need to run unnecessary migrations
cons: we must manually keep count of the migrations

##### The migration is an object

```python
migrations = [
   upgrade_v1,
   upgrade_v3,
]
for migration in migration:
  # effectively same thing as the first proposal, but instead of a check
  # inside the standalone function, it's a method
  if not upgrade_v1.applied(data):
     migration.apply()
```

pros: don't need to apply unnecessary migrations, however, this is worse then version one because the effect is the same, and version one has at most one function call per migration, while this has two
cons: Too many steps (instantiate migration, run the check, apply the migration itself)